ethcore/evm/src/interpreter/mod.rs

```diff
@@ -179,8 +181,11 @@ pub struct Interpreter<Cost: CostType> {
 	do_trace: bool,
 	done: bool,
 	valid_jump_destinations: Option<Arc<BitSet>>,
+	valid_sub_destinations: Option<Arc<BitSet>>,
 	gasometer: Option<Gasometer<Cost>>,
 	stack: VecStack<U256>,
+	return_stack: Vec<usize>,
```

Comment:
Wouldn't it be better to use platform-indenpendent `u64` instead of `usize`?

Reply:
Return stack values are internal, meaning it's not possible to inspect it other than `RETURNSUB` to it. The values of return stack are always taken from PC (program counter) directly.

Our PC is of type `usize`, so I think it makes sense to also use `usize` here. In non-64-bit system this will be slightly more efficient, but for most cases using either `usize` or `u64` does not matter (except that there will need to be an explicit conversion from PC `usize` to `u64`).

Reply:
I was more worried if that could cause consensus issues on 32-bit systems. But I guess since `PC` is `usize`, this is out of scope of this PR.

Reply:
I don't think it can. The only way it *may* cause consensus issue is if PC overflows `usize`. However, to make that happen you'd first need a code size greater than `usize`, which will exceed the architecture's memory limit (which is also addressed by `usize`).

Reply:
So if the code size is 4GB, it will just panic on a 32-bit system instead creating a consensus issue? I'm even sure openethereum works on a 32-bit system.

Reply:
Yes, because it's not possible to have code size over on a 32-bit system. That's the memory limit.

Reply:
Note, that it's possible to have more than 4GB on 32-bit with enabled PAE https://en.wikipedia.org/wiki/Physical_Address_Extension.

But I think this thread can be marked as resolved.

Reply:
The thing is -- it depends on the memory address type (in Rust that's always `usize`). So with PAE, while the memory system can use increase, a single application's addressable memory still remains at most 4GB!